storagemarket/deal_execution.go

```diff
@@ -396,6 +396,10 @@ func (p *Provider) addPiece(ctx context.Context, pub event.Emitter, deal *types.
 
 	// Add the piece to a sector
 	packingInfo, packingErr := p.AddPieceToSector(p.ctx, *deal, paddedReader)
+	if xerrors.Is(packingErr, context.Canceled) {
+		_ = v2r.Close()
+		return packingErr
+	}
```

Comment:
It's a little confusing for me to follow the logic on lines 399 - 411. Is there a way we can combine it into less if statements? Let's add some comments as well to explain what's going on.

Reply:
simplified.

Reply:
I don't see any change - do you still need to push a commit?

Reply:
pushed now.